pkg/util/azure/credential.go

```diff
@@ -129,5 +129,14 @@ func newConfigCredential(creds map[string]string, options configCredentialOption
 		return nil, errors.Errorf("%s is required", CredentialKeyPassword)
 	}
 
+	//msiEndpoint
+	msiEndpoint := creds[CredentialMSIENDPOINT]
+	identityEndpoint := creds[CredentialIDENTITYENDPOINT]
+
+	if msiEndpoint != "" || identityEndpoint != "" {
```

Comment:
How about moving this part into `NewCredential()`?
The `newConfigCredential` is a replacement for [NewEnvironmentCredential()](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/environment_credential.go#L80) in the Azure SDK and `NewCredential()` is a replacement for [NewDefaultAzureCredential()](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/default_azure_credential.go#L62).

Reply:
>> How about moving this part into NewCredential()?
the fuction NewCredential() has included the func newConfigCredential()..
What do you opinion about that? @anshulahuja98 @ywk253100

Reply:
> How about moving this part into `NewCredential()`? 

 the fuction NewCredential() has included the func newConfigCredential()..

What do you opinion about that? @anshulahuja98 @ywk253100

Reply:
Just like the [comment](https://github.com/vmware-tanzu/velero/blob/main/pkg/util/azure/credential.go#L75) of `newConfigCredential()` said: `newConfigCredential works same as the azidentity.EnvironmentCredential but reads the credentials from a map rather than environment variables`, I suggest we keep the same behavior with `EnvironmentCredential` of Azure SDK